app/src/providers/Web3Provider.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import Web3 from 'web3';
+import ArtifactRegistry from '../contracts/ArtifactRegistry.json';
+import { ContractListType } from '../helper/eth';
+
+interface Web3Interface {
+  web3: Web3;
+  accounts: Array<string>;
+  contracts: ContractListType;
+}
+
+interface Web3ProviderProps {
+  networkId: number;
+  accounts: Array<string>;
+}
+
+const Web3Context = React.createContext<Web3Interface>({} as any);
+
+export const Web3Provider: React.FC<Web3ProviderProps> = ({ networkId, accounts, children }) => {
+  const web3 = new Web3(Web3.givenProvider || 'ws://127.0.0.1:8545');
+
+  const artifactRegistryContractAddress = (ArtifactRegistry.networks as any)[networkId.toString()].address;
+  const artifactRegistry = new web3.eth.Contract(ArtifactRegistry.abi, artifactRegistryContractAddress);
+
+  // TODO: Update these to have the actual contracts when refactoring
```

Comment:
When will this refactoring happen / when will we fix this?

Reply:
Its unlikely that everything will get refactored by the end of this project in my opinion, but whenever someone needs to use the latest web3 for anything (e.g making things responsive with events) then they should also use this provider and refactor the specific class